File: paper/src/main/java/net/lib/luxorigin/LanguageAPI/LanguageAPI.java
package net.lib.luxorigin.LanguageAPI;

import net.lib.luxorigin.LanguageAPI.translate.Translate;

import java.util.Map;

final public class LanguageAPI {

    // NOOP
    private LanguageAPI() {}

    public static String translate(String key) {
        return LanguageAPIHandler
                .getCache()
                .getOrDefault(LanguageAPIHandler.getRegistered(), Map.of())
                .getOrDefault(LanguageAPIHandler.getLanguage(), Map.of())
                .getOrDefault(key, key);
    }

    public static String translate(Translate translate) {
        String text = translate(translate.getKey());

        for (var entry : translate.getTranslate().entrySet()) {
            text = text.replace("{" + entry.getKey() + "}", entry.getValue());
        }
        return text;
    }
}

File: pmmp/src/lib/luxorigin/LanguageAPI/LanguageAPI.php
<?php

declare(strict_types=1);

namespace lib\luxorigin\LanguageAPI;

use lib\luxorigin\LanguageAPI\translate\Translate;

final class LanguageAPI
{
    // NOOP
    private function __construct()
    {}

    public static function translate(Translate $translate): string
    {
        $text = LanguageAPIHandler::get($translate->getKey());

        foreach ($translate->getParameters() as $key => $value) {
            $text = str_replace('{' . $key . '}', (string)$value, $text);
        }
        return $text;
    }
}